File: src/domain/valueObjects/post-id.ts
export class PostId {
    private readonly _value: string;

    private constructor(value: string) {
        this._value = value;
    }

    get value(): string {
        return this._value;
    }

    static create(value: string): PostId {
        if (!value || value.trim().length === 0) {
            throw new Error('PostId cannot be empty');
        }
        return new PostId(value);
    }
}

File: lib/blog/domain/value_objects/post_id.dart
class PostId {
  final String value;

  PostId._({required this.value});

  factory PostId.create(String raw) {
    if (raw.trim().isEmpty) throw ArgumentError('PostId cannot be empty');
    return PostId._(value: raw.trim());
  }

  factory PostId.empty() => PostId._(value: '');

  @override
  bool operator ==(Object other) => other is PostId && other.value == value;

  @override
  int get hashCode => value.hashCode;
}
